src/main/java/org/kaazing/netx/ws/internal/io/WsInputStream.java

```diff
@@ -141,101 +145,106 @@ public int available() throws IOException {
 
     @Override
     public int read() throws IOException {
```

Comment:
I will create a private method that can be called from both the `read` calls to avoid paying extra for the lock.